### PR TITLE
Add rollable asset condition meters

### LINKED ISSUE #19: Allow meters on assets to be rolled straight from the asset

### PR DESCRIPTION
**Pull request by AG-Guardian**
Make asset condition meters rollable, additional cleanup

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ See [SETUP.md](./SETUP.md) for instructions on how to get everything running.
 
 ## Picking Up Tickets
 
-My todo list for this application can be found [here](https://github.com/users/scottbenton/projects/5).
+My todo list for this application can be found [here](https://github.com/users/scottbenton/projects/7/views/3).
```
Comment (scottbenton) [inline CONTRIBUTING.md:13]: Thanks for fixing this (and adding to the setup file)! Copied it over from the other project and forgot to update it.

File: src/data/constants.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const DEFAULT_MOMENTUM = 2;
```
Comment (scottbenton) [inline src/data/constants.ts:1]: Thanks!

File: src/components/datasworn/ConditonMeter/ConditionMeter.tsx [hunk 3]
```diff
@@ -86,6 +86,9 @@ export function ConditionMeter(props: ConditionMeterProps) {
         <Box
           id={id}
           component={onActionClick && !disabled ? ButtonBase : "div"}
+          aria-label={
+            onActionClick && !disabled ? t("datasworn.roll", "Roll") : undefined
+          }
```
Comment (scottbenton) [inline src/components/datasworn/ConditonMeter/ConditionMeter.tsx]: Everything else looks great, I'm just not positive on this one - does the aria-label on line 155 handle labeling this button, or do we need it directly on the button itself?

**Review by scottbenton — COMMENTED** [review]
Nice addition, and great cleanup, just one small question!

File: src/components/datasworn/ConditonMeter/ConditionMeter.tsx (hunk 3 above)
Comment (AG-Guardian) [reply src/components/datasworn/ConditonMeter/ConditionMeter.tsx]: I needed it directly on the button so that I could find it in the test - there is probably a better way to handle that, but yes there is an aria label on the button element as well as a nested element

File: src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx [hunk 4]
```diff
@@ -74,6 +74,19 @@ describe("AssetControl", () => {
...
+    setup({
+      control: {
+        field_type: "condition_meter",
+        label: "Condition Meter",
+        value: 5,
+        min: 0,
+        max: 10,
+      } as Datasworn.AssetControlField,
+    });
+    expect(screen.getByRole("button", { name: /Roll/i })).toBeInTheDocument();
```
Comment (AG-Guardian) [inline src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx]: ^ this is where I look for the aria label in the test - the test is making sure that an actual button element was created

File: src/components/datasworn/ConditonMeter/ConditionMeter.tsx (hunk 3 above)
Comment (scottbenton) [reply src/components/datasworn/ConditonMeter/ConditionMeter.tsx]: data-testid and `getByTestId` within the test should handle this correctly without adding the aria-label.

Comment (AG-Guardian) [reply src/components/datasworn/ConditonMeter/ConditionMeter.tsx]: done

File: src/components/datasworn/AssetCard/AssetControl.tsx [hunk 5]
```diff
@@ -98,10 +96,14 @@ export function AssetControl(props: AssetControlProps) {
             disabled={!onControlChange}
             onChange={onControlChange ? handleControlChange : undefined}
             onActionClick={onControlChange ? handleRoll : undefined}
-            action={onControlChange ? {
-              ActionIcon: RollIcon,
-              actionLabel: t("datasworn.roll", "Roll"),
-            } : undefined}
+            action={
+              onControlChange && control.rollable
```
Comment (AG-Guardian) [inline src/components/datasworn/AssetCard/AssetControl.tsx:100]: I think `control.rollable` is always true for condition meters

File: src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx [hunk 6]
```diff
@@ -74,17 +86,23 @@ describe("AssetControl", () => {
...
+        rollable: true,
+        // I would add a test case for the negative, but datasworn has this hardcoded as true currently.
       } as Datasworn.AssetControlField,
     });
-    expect(screen.getByTestId("roll-button")).toBeInTheDocument();
+    expect(mockRollStat).toHaveBeenCalledTimes(0);
+    const rollButton = screen.getByTestId("roll-button");
+    expect(rollButton).toBeInTheDocument();
+    fireEvent.click(rollButton);
+    expect(mockRollStat).toHaveBeenCalledTimes(1);
```
Comment (AG-Guardian) [inline src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx:107]: much better test

Comment (scottbenton) [reply src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx:107]: Now if only it fixed the problem lol.

Comment (AG-Guardian) [reply src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx:107]: I feel pretty confident that we need to make the env vars available to the build action - firebase.config is looking for them:
```
const firebaseConfig: FirebaseOptions = {
  apiKey: import.meta.env.VITE_FIREBASE_APIKEY,
  authDomain: import.meta.env.VITE_FIREBASE_AUTHDOMAIN,
  projectId: import.meta.env.VITE_FIREBASE_PROJECTID,
  storageBucket: import.meta.env.VITE_FIREBASE_STORAGEBUCKET,
  messagingSenderId: import.meta.env.VITE_FIREBASE_MESSAGINGSENDERID,
  appId: import.meta.env.VITE_FIREBASE_APPID,
};
```

Comment (AG-Guardian) [reply src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx:107]: just kidding the module factory worked - I guess because its preventing us from getting all the way to the firestore init

Comment (scottbenton) [reply src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx:107]: Yeah, seems like the basic vi.mock either does some kind of spy be default or imports the other modules when parsing through everything. Doing the module factory fixes that. 

Now onto the other build issue :smile: .

**Comment by scottbenton** [comment]
Build is fixed (might need to hard refresh to see the changes). Thanks for pointing that out!

**Review by scottbenton — APPROVED** [review]
